.github/actions/composite/setupNode/action.yml

```diff
@@ -9,7 +9,15 @@ runs:
         node-version-file: '.nvmrc'
         cache: npm
 
+    - id: cache-node-modules
+      if: github.run_attempt == '1'
+      uses: actions/cache@v3
+      with:
+        path: node_modules
+        key: ${{ runner.os }}-node-modules-${{ hashFiles('package-lock.json') }}
```

Comment:
Note: no partial matches are allowed, because `npm ci` will delete `node_modules` if it's present. It's either a full match and `npm ci` is skipped entirely, or it's a cache miss and we run `npm ci`. We do not do `npm i` because it can change `package-lock.json` and is considered unsafe for CI environments